File: merge-two-sorted-lists/shaun2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
#Runtime: 40 ms, faster than 23.48% of Python3 online submissions for Merge Two Sorted Lists.
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        dummy = ListNode(-1)
        cur = dummy
        p1,p2 = l1,l2
        while (p1 and p2):
            if p1.val < p2.val:
                cur.next = p1
                p1 = p1.next
            else:
                cur.next = p2
                p2 = p2.next
            cur = cur.next
        if p1 != None:
            cur.next = p1
        else:
            cur.next = p2
        return dummy.next
        

File: merge-two-sorted-lists/shaun2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
//Runtime: 0 ms, faster than 100.00% of Java online submissions for Merge Two Sorted Lists.
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode dummy = new ListNode(-1);
        ListNode cur = dummy;
        ListNode p1 = l1;
        ListNode p2 = l2;
        while (p1 != null && p2 != null) {
            if (p1.val < p2.val) {
                cur.next = p1;
                p1 = p1.next;
                
            }else {
                cur.next = p2;
                p2 = p2.next;
            }
            cur = cur.next;
        }
        if (p1 == null) {
            cur.next = p2;
        }else {
            cur.next = p1;
        }
        return dummy.next;
    }
}
